examples/toolchains/nodejs/WORKSPACE

```diff
@@ -35,11 +29,17 @@ nixpkgs_git_repository(
 )
 
 load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_nodejs_configure")
-
 nixpkgs_nodejs_configure(
+  name = "nixpkgs-nodejs_linux_amd64",
```

Comment:
As you point out, a possible fix for this may be to mimic [rules_haskell's behavior](https://github.com/tweag/rules_haskell/blob/8e0189dc8b444f81ff33e5f69ae4489cb342bab6/haskell/asterius/asterius_dependencies.bzl#L96-L113) within `nixpkgs_nodejs_configure`, or perhaps better a `nixpkgs_nodejs_configure_platforms` which invokes `nixpkgs_nodejs_configure` once for each platform.

`nixpkgs_nodejs_configure` could perhaps be changed along the lines of this (untested):
```diff
diff --git a/toolchains/nodejs/nodejs.bzl b/toolchains/nodejs/nodejs.bzl
index d4cef6b..4c20c95 100644
--- a/toolchains/nodejs/nodejs.bzl
+++ b/toolchains/nodejs/nodejs.bzl
@@ -3,7 +3,7 @@ load("@rules_nixpkgs_core//:util.bzl", "ensure_constraints")

 _nodejs_nix_content = """\
 let
-    pkgs = import <nixpkgs> {{ config = {{}}; overlays = []; }};
+    pkgs = import <nixpkgs> {{ config = {{}}; overlays = []; system = {system}; }};
     nodejs = pkgs.{attribute_path};
 in
 pkgs.buildEnv {{
@@ -74,6 +74,7 @@ def nixpkgs_nodejs_configure(
   nixopts = [],
   fail_not_supported = True,
   quiet = False,
+  system = None,
   exec_constraints = None,
   target_constraints = None,
 ):
@@ -83,6 +84,7 @@ def nixpkgs_nodejs_configure(
     if not nix_file and not nix_file_content:
       nix_file_content = _nodejs_nix_content.format(
         attribute_path = attribute_path,
+        system = repr(system) if system else "builtins.currentSystem",
       )

     nixpkgs_package(
```
Callers that set `system` must also set `exec|target_constraints` accordingly.

Reply:
Why do we need this (additional `system` param)? I probably know it's not targeted to remote build execution, but isn't enough just to take all the platforms that rules_nodejs defines (yes... I know it's pointless e.g. in case of windows) and iterate over them (like I did it in my next commit)?

Reply:
> isn't enough just to take [all the platforms that rules_nodejs defines](https://github.com/bazelbuild/rules_nodejs/blob/a5755eb458c2dd8e0e2cf9b92d8304d9e77ea117/nodejs/private/toolchains_repo.bzl#L20) [...] and iterate through them

In that case the platform in the workspace name does not match the platform for which the contained nix-import works. E.g. if you build on x64 Linux `nixpkgs-nodejs_macos_arm64` would still give you an x64 Linux nodejs installation, because `system` defaults to `builtins.currentSystem`, so when this is evaluated on x64 Linux, then that's what you get.

It works for local builds in the case of it being a workaround to the naming requirement. But, it's pretty misleading. (For that route it would be better to stick to a single `nixpkgs_nodejs_configure` and select the name based on the host platform. Or, perhaps use the [implicit `repo_mapping` attribute](https://bazel.build/reference/be/workspace#local_repository.repo_mapping) on `npm_install` - if that works.)

Reply:
Ok, so I have next question.

When you evaluate your build on x86-64_linux, why you want to get nodejs for e.g. macos (or any other architecture)? Maybe it's a general question, why one evaluating build on architecture A, would like to execute (on architecture A) the binary compiled for architecture B? Why in this case (nodejs) we **don't** want to have 1:1 matching between evaluation architecture and node binary "version".

Sorry - it sounds elementary, but this answer would give me better understanding of what we're doing here.

Reply:
> When you evaluate your build on x86_64-linux, why you want to get nodejs for e.g. macos (or any other architecture)?

That would only be relevant in a remote execution setting, you can find more details [here](https://github.com/tweag/rules_nixpkgs/issues/180#issuecomment-1284656149) and [here](https://github.com/tweag/rules_nixpkgs/issues/180#issuecomment-1381090829). In brief, if you evaluate on a Linux host to build on a MacOS remote executor, then you'd need the MacOS nodejs.

However, as described in https://github.com/tweag/rules_nixpkgs/issues/180, remote execution is not well supported by rules_nixpkgs, yet. So, we don't need to make support for it a blocker for this issue.

My point is more about not generating misleading external workspaces. The problem with the [approach here](https://github.com/tweag/rules_nixpkgs/pull/309/commits/814b9c45c7face4c590c2b9bfd12de8e74c666fc#diff-150d5ce7d9b192ca069368c968ea722a87abd8ca74b98149f0b54e1950c9d60eR115) is that it generates multiple copies of the same external workspace under different names such that on an x64 Linux host both, `nixpkgs_nodejs_linux_amd64` and `nixpkgs_nodejs_macos_arm64`, generate an x64 Linux nodejs. That is bound to cause confusion down the line.

So, if we generate multiple workspaces, then they should be correct, i.e. `nixpkgs_nodejs_linux_amd64` always gives you an x64 Linux nodejs, and `nixpkgs_nodejs_macos_arm64` always gives you an arm64 MacOS nodejs. Or, alternatively, we only generate a workspace for the current host.

Reply:
Thank you for this answer. I tried to address this issue (probably in a weak way). I am fixing `target|exec_constraints` to the ones that are defined under `rules_nodejs`. Please take a look.